media_driver/linux/common/os/mos_gpucontext_specific.cpp

```diff
@@ -112,6 +115,123 @@ MOS_STATUS GpuContextSpecific::Init(OsContext *osContext)
+                sseu.engine.engine_instance = 0;
+                sseu.flags = I915_CONTEXT_SSEU_FLAG_ENGINE_INDEX;
+
+                if (mos_get_context_param_sseu(m_i915Context, &sseu))
+                {
+                    MOS_OS_ASSERTMESSAGE("Failed to get sseu configuration.");
+                    return MOS_STATUS_UNKNOWN;
+                };
+
+                if (mos_hweight8(sseu.subslice_mask) > createOption->packed.SubSliceCount)
+                {
+                    sseu.subslice_mask = mos_switch_off_n_bits(sseu.subslice_mask,
+                            mos_hweight8(sseu.subslice_mask)-createOption->packed.SubSliceCount);
+                }
+
+                if (mos_set_context_param_sseu(m_i915Context, sseu))
+                {
+                    MOS_OS_ASSERTMESSAGE("Failed to set sseu configuration.");
+                    return MOS_STATUS_UNKNOWN;
+                };
+            }
+        }
+        else if (GpuNode == MOS_GPU_NODE_VIDEO || GpuNode == MOS_GPU_NODE_VIDEO2)
+        {
+            unsigned int nengine = MAX_ENGINE_INSTANCE_NUM;
+            struct i915_engine_class_instance engine_map[MAX_ENGINE_INSTANCE_NUM];
+            __u16 engine_class = I915_ENGINE_CLASS_VIDEO;
+            __u64 caps = 0;
+
+            if (m_createOptionEnhanced->UsingSFC)
+            {
+                caps |= I915_VIDEO_AND_ENHANCE_CLASS_CAPABILITY_SFC;
+            }
+
+            MOS_ZeroMemory(engine_map, sizeof(engine_map));
+            if (mos_query_engines(osInterface->pOsContext->fd,engine_class,caps,&nengine,engine_map))
+            {
+                MOS_OS_ASSERTMESSAGE("Failed to query engines.\n");
+                return MOS_STATUS_UNKNOWN;
+            }
+
+            if (mos_set_context_param_load_balance(m_i915Context, engine_map, nengine))
+            {
+                MOS_OS_ASSERTMESSAGE("Failed to set balancer extension.\n");
+                return MOS_STATUS_UNKNOWN;
+            }
+        }
+        else if (GpuNode == MOS_GPU_NODE_VE)
+        {
+            unsigned int nengine = MAX_ENGINE_INSTANCE_NUM;
+            struct i915_engine_class_instance engine_map[MAX_ENGINE_INSTANCE_NUM];
+            __u16 engine_class = I915_ENGINE_CLASS_VIDEO_ENHANCE;
+            __u64 caps = 0;
+
+            if (m_createOptionEnhanced->UsingSFC)
+            {
+                caps |= I915_VIDEO_AND_ENHANCE_CLASS_CAPABILITY_SFC;
+            }
+
+            MOS_ZeroMemory(engine_map, sizeof(engine_map));
+            if (mos_query_engines(osInterface->pOsContext->fd,engine_class,caps,&nengine,engine_map))
+            {
+                MOS_OS_ASSERTMESSAGE("Failed to query engines.\n");
+                return MOS_STATUS_UNKNOWN;
+            }
+
+            if (mos_set_context_param_load_balance(m_i915Context, engine_map, nengine))
+            {
+                MOS_OS_ASSERTMESSAGE("Failed to set balancer extension.\n");
+                return MOS_STATUS_UNKNOWN;
+            }
+        }
```

Comment:
Looks like the MOS_GPU_NODE_VIDEO and MOS_GPU_NODE_VE branches are completely identical apart from the engine class. So you could consolidate them into one branch and just toggle the class based on GpuNode.

Reply:
yes, fixed.